thinc/backends/ops.py

```diff
@@ -680,6 +682,167 @@ def backprop_relu(
         dY *= Y > 0
         return dY
 
+    def relu_n(
+        self, X: FloatsType, n: float = 6.0, inplace: bool = False
+    ) -> FloatsType:
+        if inplace:
+            self.xp.clip(X, 0, n, out=X)
+            return X
+        return self.xp.clip(X, 0, n)
+
+    def backprop_relu_n(
+        self, dY: FloatsType, X: FloatsType, n: float = 6.0, inplace: bool = False
+    ) -> FloatsType:
+        if inplace:
+            dY *= (0 < X) & (X < n)
```

Comment:
One operation less, though I don't know if it is faster:

```
dY *= self.xp.clip(X, 0, n) != 0.
```

Which is of course:

```
dY *= relu_n(X) != 0.
```

Reply:
Oh I think the `clip` in the backprop is not correct, because it should be strictly `<` and `>` and not `<=` and `=<` like `clip` does.

Reply:
I would be surprised if it matters in practice. Weights are rarely integers.

Reply:
I doesn't but `hypothesize` fails the test with the evil value of `6.0`

Reply:
Ah, I see! 😀